src/main/resources/errors.yaml

```diff
@@ -45,31 +45,125 @@
 # NOTE: Please keep the entries sorted on their name for snippets, or scope and code for errors. Please add a
 #       new line after each entry, using the `|-` to trim trailing newlines.
 
+# ================================================================================================================
+# Snippets
+# ================================================================================================================
+
 snippets:
   - name: CONTACT_SUPPORT
     body: |-
       Please contact support if the issue persists.
 
-request-errors:
-  - scope: FILTER
-    code: MULTIPLE_ID_FILTER
-    title: the title for the error
+  - name: RETRY
     body: |-
-      big long message with ${vars} in it
+      It is safe to retry this request.
+
+# ================================================================================================================
+# Request Errors
+# ================================================================================================================
+
+request-errors:
+  # UNSCOPED request errors
   - scope:
     code: DOCUMENT_UNPARSEABLE
     title: Unable to parse the document
     body: |-
       big long message with ${vars} in it.
-        
+      
+      ${SNIPPET.CONTACT_SUPPORT}
+
+ # DOCUMENT request errors
+  - scope: DOCUMENT
+    code: MISSING_PRIMARY_KEY_COLUMNS
+    title: Missing table primary key columns in document
```

Comment:
q: Do we not want to refer to "Tables" here (over Document)? (I'm sure we can tweak this later on)

Reply:
in this case it only applies to a Table, and do not have a common language to explain are we inserting a "document into a table" or a "row into a table" ? 

but yes, all this is so we can finesse easier :)

Reply:
Yeah, bit too early for bike shedding I realized.